### DISCUSSION TITLE
Matching Pattern for Email Templates

### DISCUSSION THREAD

**Discussion by rwieruch** [resend/react-email#1378] [Ideas]
At the moment, the email template dev server looks for the /emails folder. We can define a `--dir`, however, is it also possible to define a file matching pattern so that I can just point the --dir to `src/`? Reason: I would like to nest the emails accordingly in my feature folders.

**Comment by gabrielmfern** [comment]
I'm so sorry, could not follow what you mean with this. Could you rephrase it?

**Comment by rwieruch** [comment]
Sorry for being to brief here. What I would want is a feature folder structure like the following:

```
src/
--/components
--/features
----/authentication
------/components
------/queries
------/actions
------/emails
--------/email-password-reset.tsx
--------/email-organization-invite.tsx
------/hooks
----/posts
------/components
------/queries
------/actions
------/emails
--------/email-post-created.tsx
--------/email-post-deleted.tsx
------/hooks
```

If I run `email": "email dev"`, all these emails in the features folders are not picked up. I could define a dir like:

```
"email": "email dev --dir src/emails"
```

But this does not help me. I would want to define a matching pattern instead:

```
"email": "email dev --match src/**/emails/*.tsx"
```

**Reply by gabrielmfern** [reply]
Hmm, yeah, I get it now. We don't really have anything that allows for this, this might be an interesting idea if anyone else is interested, but it's a pretty specific use case here as far as I can see. I'll convert the discussion to an idea one.

**Reply by rwieruch** [reply]
Understood. I think this becomes a thing in larger applications where you do not have technical slices anymore. Same when it is not viable anymore to have a *tests/* top-level folder, because you want to have them co-located to your feature folders. I know that Jest went through the same development for their matching patterns several years ago. But anyway, thanks for the consideration!
